packages/material-ui-lab/src/Rating/Rating.test.js

```diff
@@ -110,4 +111,70 @@ describe('<Rating />', () => {
     checked = container.querySelector('input[name="rating-test"]:checked');
     expect(checked.value).to.equal('2');
   });
+
+  describe('<form> integration', () => {
+    before(function beforeHook() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        // JSDOM has issues with form validation for certain elements.
+        // We could adress them individually but that doesn't add much value if we already have a working environment.
+        this.skip();
+      }
+    });
+
+    [
+      {
+        ratingProps: { name: 'rating', defaultValue: 2 },
+        formData: [['rating', '2']],
+      },
+      {
+        ratingProps: { name: 'rating', defaultValue: 2, disabled: true },
+        formData: [],
+      },
+      {
+        ratingProps: { name: 'rating', defaultValue: 2, readOnly: true },
+        // FIXME: With native <input type="radio" /> read-only values are still submitted
+        // formData: [['rating', '2']],
+        formData: [],
+      },
+      {
+        ratingProps: { name: 'rating', required: true },
+        // FIXME: `Rating` does not implement `required`.
+        //        Native <input type="radio" /> would not pass validation
+        // formData: undefined,
+        formData: [['rating', '']],
+      },
+    ].forEach((testData, testNumber) => {
+      it(`submits the expected form data #${testNumber + 1}`, () => {
+        /**
+         * @type FormData
+         */
+        let data;
+        render(
+          <form
+            onSubmit={(event) => {
+              // Prevent navigation
+              event.preventDefault();
+              // populate FormData with the submitted form
+              data = new FormData(event.target);
+            }}
+          >
+            <Rating {...testData.ratingProps} />
+            <button type="submit" />
+          </form>,
+        );
+        const submitter = document.querySelector('button[type="submit"]');
+
+        act(() => {
+          // form.submit() would not run form validation
+          submitter.click();
+        });
+
+        if (testData.formData === undefined) {
```

Comment:
Does the test ever enter in this branch?

Reply:
Not at the moment. I technically should for `<input type="radio" required />` because the form wouldn't be submitted (which is what `formData: undefined` represents) but that's not how `<Rating required />` works at the moment. 

I'll add a comment explaining that overload.